cmd/arrai/eval_test.go

```diff
@@ -70,3 +70,37 @@ func TestEvalCondExpr(t *testing.T) {
 	assertEvalExprString(t, "(*:(1+2))", "cond (*: 1 + 2)")
 	assertEvalExprString(t, "((1<2):1,*:(1+2))", "cond (1 < 2: 1, * : 1 + 2)")
 }
+
+func TestEvalCondWithControlVar(t *testing.T) {
+	t.Parallel()
+	// Control var conditions
+	assertEvalOutputs(t, `1`, `let a = 1; a cond (1 :1, 2 :2, *:1 + 2)`)
+	assertEvalOutputs(t, `11`, `let a = 1; a cond (1 :1 + 10, 2 : 2, *:1 + 2)`)
+	assertEvalOutputs(t, `3`, `let a = 1; a cond (2 :2, *:1 + 2)`)
+	assertEvalOutputs(t, `13`, `let a = 1; a cond (2 :2, *:11 + 2)`)
+	assertEvalOutputs(t, `5`, `let a = 3; a cond (*:5)`)
+	assertEvalOutputs(t, `3`, `let a = 3; a cond (*:1 + 2)`)
+
+	assertEvalOutputs(t, `1`, `let a = 1; let b = a cond (1 :1, 2 :2, *:1 + 2); b`)
+	assertEvalOutputs(t, `100`, `let a = 1; let b = a cond (1 :1, 2 :2, *:1 + 2); b * 100`)
+	//
+	assertEvalOutputs(t, `2`, `let a = 1; (a + 1) cond (1 :1, 2 :2, *:1 + 2)`)
+	assertEvalOutputs(t, `3`, `let a = 1; (a + 10) cond (1 :1, 2 :2, *:1 + 2)`)
+	assertEvalOutputs(t, `2`, `let a = 1; let b = (a + 1) cond (1 :1, 2 :2, *:1 + 2); b`)
+	assertEvalOutputs(t, `300`, `let a = 1; let b = (a + 10) cond (1 :1, 2 :2, *:1 + 2); b * 100`)
+
+	var sb strings.Builder
+	assert.Error(t, evalImpl(`let a = 3; a cond (1 :1, 2 :2 + 1)`, &sb))
+	assert.Error(t, evalImpl(`let a = 3; let b = a cond (1 :1, 2 :2 + 1); b`, &sb))
+	assert.Error(t, evalImpl(`let a = 3; let b = (a + 10) cond (1 :1, 2 :2 + 1); b`, &sb))
+}
+
+func TestEvalCondWithControlVarExpr(t *testing.T) {
+	t.Parallel()
+	assertEvalExprString(t, "(1->(\\a((control_var:a),((1+2):1,*:(1+2)))))",
```

Comment:
Use backquotes to avoid the need for `\\`.

Reply:
`\\` is generated by `struct Function` in go file value_set_func.go, the method `func (f *Function) String() string`. The syntax is `let`.